### PR TITLE
Improve client cache metrics

### PR DESCRIPTION
**Pull request by calvinjia**
Adds bytes evicted and bytes written. Also makes the LocalCacheManager tolerant of page store failures.

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 1]
```diff
@@ -268,4 +261,65 @@ public void delete(PageId pageId) throws IOException, PageNotFoundException {
...
+    } catch (IOException | PageNotFoundException e) {
+      LOG.error("Failed to delete page {}: {}", pageId, e);
+      return false;
+    }
+    mEvictor.updateOnDelete(pageId);
+    Metrics.BYTES_EVICTED_CACHE.inc(pageInfo.getPageSize());
+    return true;
+  }
+
+  private ReadableByteChannel getPage(PageId pageId, int offset) {
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java]: Add `@Nullable`

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java [hunk 2]
```diff
@@ -164,13 +167,13 @@ public boolean put(PageId pageId, byte[] page) throws IOException {
           victimPageInfo = mMetaStore.getPageInfo(victim);
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java]: move this down to be protected by lock as discussed

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java (hunk 1 above)
Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java]: It is annotated in the interface, is that sufficient?

Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java]: Updated, thanks.

File: core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java [hunk 3]
```diff
@@ -33,45 +32,41 @@ static CacheManager create(AlluxioConfiguration conf) throws IOException {
...
 
   /**
    * Deletes a page from the cache.
    *
    * @param pageId page identifier
-   * @throws PageNotFoundException if page is not found in the store
+   * @return true if the page is successfully deleted, false otherwise
    */
-  void delete(PageId pageId) throws IOException, PageNotFoundException;
+  boolean delete(PageId pageId) throws IOException;
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java]: still leave IOE here o signature?

**Comment by calvinjia** [comment]
jenkins, test this please.

File: core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java (hunk 3 above)
Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java]: Updated, thanks.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
alluxio-bot, cherry-pick this to branch-2.2 please

**Pull request merged**